### PR TITLE
feat(auth): edit login flow

### PR DESCRIPTION
**Pull request by wass3r**
implements https://github.com/go-vela/community/issues/81

changes to web based auth flow

removed the login tests for now. since we are opening a browser window, i didn't see a great way to test. the only thing you need is the addr, which is checked already elsewhere. both GH's fork of `pkg/browser` (https://github.com/cli/browser) and the original didn't have tests (https://github.com/pkg/browser) afaict.

not addressing `lll` and `dupl` linter issues in test files as those will be addressed separately at some point.

### CODE REVIEW

File: internal/client/client_test.go [hunk 1]
```diff
@@ -91,7 +107,7 @@
 
 	// run tests
 	for _, test := range tests {
-		_, err := ParseEmptyToken(cli.NewContext(nil, test.set, nil))
+		_, err := ParseEmptyToken(cli.NewContext(&cli.App{Name: "vela", Version: "v0.0.0"}, test.set, nil))
```
[the remark this answers is not in the document]
Comment (wass3r) [reply internal/client/client_test.go:110]: ignoring.. will be address when change to linter to ignore `lll` and `dupl` for test files comes in

File: action/completion_generate_test.go [hunk 2]
```diff
@@ -40,7 +40,7 @@
 
 	// run tests
 	for _, test := range tests {
-		err := completionGenerate(cli.NewContext(nil, test.set, nil))
+		err := completionGenerate(cli.NewContext(&cli.App{Name: "vela", Version: "v0.0.0"}, test.set, nil))
```
[the remark this answers is not in the document]
Comment (wass3r) [reply action/completion_generate_test.go:43]: ignoring.. will be address when change to linter to ignore `lll` and `dupl` for test files comes in

**Review by kneal — DISMISSED** [review]
LGTM 🐬

**Review by jbrockopp — APPROVED** [review]
LGTM

**Review by jbrockopp — APPROVED** [review]
LGTM

**Review by kneal — APPROVED** [review]
LGTM 🐬

**Pull request merged**